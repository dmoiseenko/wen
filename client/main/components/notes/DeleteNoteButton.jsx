import React from 'react';
import PropTypes from 'prop-types';


export default function DeleteNoteButton({ deleteNote }) {
  function handleClick() {
    deleteNote();
  }

  return (
    <a
      className="fa fa-times note__delete"
      onClick={handleClick}
    />
  );
}

DeleteNoteButton.propTypes = {
  deleteNote: PropTypes.func.isRequired,
};
