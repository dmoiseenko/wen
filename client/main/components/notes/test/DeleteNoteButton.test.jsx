import React from 'react';
import { shallow } from 'enzyme';

import DeleteNoteButton from '../DeleteNoteButton';


it('should render', () => {
  const props = {
    deleteNote: jest.fn()
  };

  const wrapper = shallow(
    <DeleteNoteButton {...props} />
  );

  expect(wrapper.node).toMatchSnapshot();
});

it('should call deleteNote when click fires', () => {
  const deleteNote = jest.fn();

  const wrapper = shallow(
    <DeleteNoteButton
      deleteNote={deleteNote}
    />
  );

  wrapper.find('.note__delete').simulate('click');

  expect(deleteNote).toHaveBeenCalled();
});
