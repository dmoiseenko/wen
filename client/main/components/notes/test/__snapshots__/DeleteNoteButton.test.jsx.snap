// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<a
  className="fa fa-times note__delete"
  onClick={[Function]}
/>
`;
